### PR TITLE
added pending update list and Apple Silicon support

### PR DESCRIPTION
**Pull request by mpanighetti**
- added comma-separated list of pending software updates to display in user-facing dialog boxes
- added basic Apple Silicon architecture support #45
  - script prompts for manual update action, opens System Preferences - Software Update, and repeats until no recommended updates are detected or the system is restarted
- moved `check_for_updates` run before `FORCE_DATE` check (prevents the deferral countdown clock from starting until after update cache has finished on first script run)
- switched default logo path to use `/System/Library/PreferencePanes/SoftwareUpdate.prefPane/Contents/Resources/SoftwareUpdate.icns` (Big Sur does not have an icon for the Software Update app)

### CODE REVIEW

File: payload/Library/Scripts/Install or Defer.sh [hunk 1]
```diff
@@ -421,6 +428,9 @@ else
 fi
 echo "Maximum deferral time: $(convert_seconds "$MAX_DEFERRAL_TIME")"
 
+# Check for updates, exit if none found, otherwise cache locally and continue.
+check_for_updates
```
Comment (mpanighetti) [inline payload/Library/Scripts/Install or Defer.sh:432]: This change addresses an extreme edge case: I found that my update test Macs, which have a custom deferral period of 1 hour, were running down the clock by defining the start time before starting the cache download. macOS updates are only getting bigger, and as more folks are currently working away from corporate offices, they will potentially be running this script on slower home networks and/or VPN, so I think it makes sense to not start that countdown until the update is cached for the first time.

File: payload/Library/Scripts/Install or Defer.sh [hunk 2]
```diff
@@ -190,31 +226,61 @@ EOF
...
-        if [[ "$UPDATE_OUTPUT_CAPTURE" == *"select Shut Down from the Apple menu"* ]]; then
-            trigger_restart "shut down"
-        # Otherwise, restart the Mac.
-        else
-            trigger_restart "restart"
+        echo "This is an Apple Silicon Mac with pending updates. Displaying persistent alert until updates are applied..."
+
+        # Loop this check until softwareupdate --list shows no more pending
+        # recommended updates.
+        while [[ $(/usr/sbin/softwareupdate --list) == *"Recommended: YES"* ]]; do
```
Comment (mpanighetti) [inline payload/Library/Scripts/Install or Defer.sh:238]: Open to discussion on possibly adding a delay to this while loop. Maybe a `sleep 60` before `killall jamfHelper`? Gives people a chance to save documents and quit apps before System Preferences reasserts itself.

**Review by homebysix — APPROVED** [review]

File: payload/Library/Scripts/Install or Defer.sh (hunk 2 above)
Comment (mpanighetti) [reply payload/Library/Scripts/Install or Defer.sh:238]: Added `sleep 60` below.
